pkg/templates/canal/daemonset.go

```diff
@@ -66,8 +67,11 @@ func daemonSet(ifacePatch bool) *appsv1.DaemonSet {
 		flannelEnv = append(flannelEnv, corev1.EnvVar{
 			Name: "FLANNELD_IFACE",
 			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "status.hostIP",
+				ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+					Key: "canal_iface",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "canal-config",
+					},
```

Comment:
this change will break cluster with `external: true`

Reply:
see cb97908a3825e72484b15ed446216e17d851c001 for more info

Reply:
@kron4eg I've already reverted it, please see the latest commit.